File: Scov/src/Scov/module/impl/player/FastUse.java
package Scov.module.impl.player;

import Scov.api.annotations.Handler;
import Scov.events.player.EventMotionUpdate;
import Scov.module.Module;
import Scov.value.impl.BooleanValue;
import Scov.value.impl.EnumValue;
import net.minecraft.client.entity.EntityPlayerSP;
import net.minecraft.client.settings.GameSettings;
import net.minecraft.item.Item;
import net.minecraft.item.ItemFood;
import net.minecraft.item.ItemPotion;
import net.minecraft.item.ItemStack;
import net.minecraft.network.play.client.C03PacketPlayer;

public class FastUse extends Module {
	
	private EnumValue<Mode> mode = new EnumValue("FastUse Mode", Mode.NCP);
	
	private BooleanValue groundOnly = new BooleanValue("GroundOnly", true);
	
	public FastUse() {
		super("FastUse", 0, ModuleCategory.PLAYER);
		addValues(mode, groundOnly);
	}
	
	@Handler
	public void onMotionUpdate(final EventMotionUpdate event) {
		EntityPlayerSP player = mc.thePlayer;
		GameSettings gameSettings = mc.gameSettings;
		Item item = player.getHeldItem().getItem();
		switch (mode.getValue()) {
		case NCP:
	        if (item != null && (item instanceof ItemFood || item instanceof ItemPotion) && gameSettings.keyBindDrop.isKeyDown() && (!groundOnly.getValue() || player.onGround) && player.ticksExisted % 4 == 0) {
	            for (int i = 0; i < 2; ++i) {
	                player.sendQueue.addToSendQueue(new C03PacketPlayer.C06PacketPlayerPosLook(player.posX, player.posY + 1.0E-9, player.posZ, player.rotationYaw, player.rotationPitch, true));
	            }
	        }
			break;
		case Vanilla:
	        if (player.getCurrentEquippedItem() != null && (player.getCurrentEquippedItem().getItem() instanceof ItemFood || player.getCurrentEquippedItem().getItem() instanceof ItemPotion) && gameSettings.keyBindDrop.isKeyDown() && (!this.groundOnly.getValue() || player.onGround)) {
	            for (int i = 0; i < 9; ++i) {
	                player.sendQueue.addToSendQueue(new C03PacketPlayer.C06PacketPlayerPosLook(player.posX, player.posY, player.posZ, player.rotationYaw, player.rotationPitch, true));
	            }
	        }
			break;
		}
	}
	
	public void onEnable() {
		super.onEnable();
	}
	
	public void onDisable() {
		super.onDisable();
	}
	
	private enum Mode {
		NCP, Vanilla
	}
}

File: Helium/rip/helium/cheat/impl/player/FastUse.kt
package rip.helium.cheat.impl.player

import me.hippo.systems.lwjeb.annotation.Collect
import net.minecraft.item.ItemFood
import net.minecraft.item.ItemPotion
import rip.helium.cheat.Cheat
import rip.helium.cheat.CheatCategory
import rip.helium.event.minecraft.PlayerUpdateEvent
import net.minecraft.network.play.client.C03PacketPlayer
import net.minecraft.network.play.client.C03PacketPlayer.C06PacketPlayerPosLook
import rip.helium.ChatUtil
import rip.helium.utils.property.impl.DoubleProperty
import rip.helium.utils.property.impl.StringsProperty


/*/
Created by Kansio on 1/16/2021
 */

class FastUse : Cheat("FastUse", "Uses shit faster", CheatCategory.PLAYER) {


    var fastuseMode = StringsProperty("Mode", "How the priority target will be selected.", null, false, true, arrayOf("Vanilla", "Ghostly"), arrayOf(true, false));

    @Collect
    fun onPlayerUpdate(event: PlayerUpdateEvent) {
        when (fastuseMode.selectedStrings.get(0)) {
            "Vanilla" -> {
                if (mc.thePlayer.isUsingItem) {
                    if (mc.thePlayer.onGround) {
                        if (isUsingItem()) {
                            repeat(20) {
                                mc.thePlayer.sendQueue.addToSendQueue(C03PacketPlayer(true));
                            }
                        }
                    }
                }
            }
            "Ghostly" -> {
                if (mc.thePlayer.isUsingItem) {
                    if (mc.thePlayer.onGround) {
                        if (this.mc.thePlayer.ticksExisted % 4 == 0) {
                            repeat(8) {
                                val d = mc.thePlayer.posX; Double
                                val d2 = mc.thePlayer.posY + 1.0E-9; Double
                                val d3 = mc.thePlayer.posZ; Double
                                mc.thePlayer.sendQueue.addToSendQueue(C06PacketPlayerPosLook(d, d2, d3, mc.thePlayer.rotationYaw, mc.thePlayer.rotationPitch, true))
                            }
                        }
                    }
                }
            }
        }
    }

    init {
        registerProperties(fastuseMode);
    }

    fun isUsingItem(): Boolean {
        return mc.thePlayer.currentEquippedItem != null && (this.mc.thePlayer.currentEquippedItem.item is ItemFood || this.mc.thePlayer.currentEquippedItem.item is ItemPotion) && mc.thePlayer.onGround
    }

}